src/main/java/uk/ac/cam/cl/dtg/isaac/api/PagesFacade.java

```diff
@@ -490,46 +492,59 @@ public final Response getTopicSummaryPage(@Context final Request request,
-        }
-
         try {
-            AbstractSegueUserDTO user = userManager.getCurrentUser(httpServletRequest);
-            Map<String, Map<String, List<QuestionValidationResponse>>> userQuestionAttempts;
-            try {
-                userQuestionAttempts = questionManager.getQuestionAttemptsByUser(user);
-            } catch (SegueDatabaseException e) {
-                String message = "SegueDatabaseException whilst trying to retrieve user question attempt data";
-                log.error(message, e);
-                return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR, message).toResponse();
-            }
+            // Load the summary page:
+            Content contentDOById = this.contentManager.getContentDOById(this.contentManager.getCurrentContentSHA(), summaryPageId);
+            ContentDTO contentDTOById = this.contentManager.getContentById(this.contentManager.getCurrentContentSHA(), summaryPageId);
 
-            Response result = this.findSingleResult(fieldsToMatch, userQuestionAttempts);
+            if (!(contentDOById instanceof IsaacTopicSummaryPage && contentDTOById instanceof IsaacTopicSummaryPageDTO)) {
+                return SegueErrorResponse.getResourceNotFoundResponse(String.format(
+                        "Unable to locate topic summary page with id: %s", summaryPageId));
+            }
+            IsaacTopicSummaryPage topicSummaryDO = (IsaacTopicSummaryPage) contentDOById;
+            IsaacTopicSummaryPageDTO topicSummaryDTO = (IsaacTopicSummaryPageDTO) contentDTOById;
 
-            // If we have a topic summary, log the request:
-            Object resultEntity = result.getEntity();
-            if (resultEntity instanceof IsaacTopicSummaryPageDTO) {
-                IsaacTopicSummaryPageDTO topicSummaryPageDTO = (IsaacTopicSummaryPageDTO) resultEntity;
+            AbstractSegueUserDTO user = userManager.getCurrentUser(httpServletRequest);
+            Map<String, Map<String, List<QuestionValidationResponse>>> userQuestionAttempts;
 
-                // Log the request:
-                ImmutableMap<String, String> logEntry = new ImmutableMap.Builder<String, String>()
-                        .put(PAGE_ID_LOG_FIELDNAME, summaryPageId)
-                        .put(CONTENT_VERSION_FIELDNAME, this.contentManager.getCurrentContentSHA()).build();
-                getLogManager().logEvent(user, httpServletRequest, IsaacLogType.VIEW_TOPIC_SUMMARY_PAGE, logEntry);
+            // Augment related questions with attempt information:
+            userQuestionAttempts = questionManager.getQuestionAttemptsByUser(user);
+            this.augmentContentWithRelatedContent(this.contentIndex, topicSummaryDTO, userQuestionAttempts);
+
+            // Augment linked gameboards using the list in the DO:
+            // FIXME: this requires loading both the DO and DTO separately, since augmenting things is hard right now.
```

Comment:
I think the nicer fix is likely to involve more work and adding special content mapping rules/deserialisers under ContentMapper.java generateNewPreconfiguredContentMapper.

This will do for now though